src/components/main/main-layout.tsx

```diff
@@ -1,20 +1,533 @@
+import { HeroLogic } from '../../logic/hero-logic';
+import { HeroStateModal } from '../modals/hero-state/hero-state-modal';
+import type { Item } from '../../models/item';
+import { ItemModal } from '../modals/item/item-modal';
+import type { Kit } from '../../models/kit';
+import { KitModal } from '../modals/kit/kit-modal';
+import type { MonsterGroup } from '../../models/monster';
+import { MonsterGroupModal } from '../modals/monster-group/monster-group-modal';
+import { MonsterModal } from '../modals/monster/monster-modal';
+import type { Perk } from '../../models/perk';
+import { PerkModal } from '../modals/perk/perk-modal';
+import { RulesModal } from '../modals/rules/rules-modal';
+import type { Sourcebook } from '../../models/sourcebook';
+import type { SourcebookElementKind } from '../../models/sourcebook-element-kind';
+import { SourcebookLogic } from '../../logic/sourcebook-logic';
+import { SourcebooksModal } from '../modals/sourcebooks/sourcebooks-modal';
+import type { Title } from '../../models/title';
+import { TitleModal } from '../modals/title/title-modal';
+import { Utils } from '../../utils/utils';
 import pbds from '../../assets/powered-by-draw-steel.png';
+import { useNavigation } from '../../hooks/use-navigation';
+import { usePersistedHeroes } from '../../hooks/use-persisted-heroes';
+import { usePersistedPlaybook } from '../../hooks/use-persisted-playbook';
+import { usePersistedSourcebooks } from '../../hooks/use-persisted-sourcebooks';
 
 interface Props {
-	section: 'hero' | 'library' | 'encounter';
-	drawer: ReactNode;
-	setDrawer: React.Dispatch<React.SetStateAction<ReactNode>>;
+	onAncestryCreate: (ancestry: Ancestry, sourcebook: Sourcebook | null) => void;
+	onCareerCreate: (culture: Career, sourcebook: Sourcebook | null) => void;
+	onClassCreate: (heroClass: HeroClass, sourcebook: Sourcebook | null) => void;
+	onComplicationCreate: (complication: Complication, sourcebook: Sourcebook | null) => void;
+	onCultureCreate: (culture: Culture, sourcebook: Sourcebook | null) => void;
+	onDomainCreate: (domain: Domain, sourcebook: Sourcebook | null) => void;
+	onItemCreate: (item: Item, sourcebook: Sourcebook | null) => void;
+	onKitCreate: (kit: Kit, sourcebook: Sourcebook | null) => void;
+	onPerkCreate: (perk: Perk, sourcebook: Sourcebook | null) => void;
+	onTitleCreate: (title: Title, sourcebook: Sourcebook | null) => void;
+	onEncounterDelete: (encounterId: string) => Promise<void> | void;
+	onHeroChange: (hero: Hero) => Promise<void> | void;
+	onMonsterGroupCreate: (monsterGroup: MonsterGroup, sourcebook: Sourcebook | null) => void;
 }
-
-export const MainLayout = (props: Props) => {
+export const MainLayout = ({
+	onAncestryCreate,
+	onCareerCreate,
+	onClassCreate,
+	onComplicationCreate,
+	onCultureCreate,
+	onDomainCreate,
+	onItemCreate,
+	onKitCreate,
+	onPerkCreate,
+	onTitleCreate,
+	onEncounterDelete,
+	onHeroChange,
+	onMonsterGroupCreate
+}: Props) => {
```

Comment:
keeping the props destruct here; apparently it's recommended for including specific props in `deps` of `useCallback` (according to the lint)